src/main/java/edu/cornell/kfs/module/purap/document/web/struts/CuPaymentRequestAction.java

```diff
@@ -69,7 +74,25 @@ public int compare(Object o1, Object o2) {
 	    }
         return forward;
     }
-	
+
+    @Override
+    protected void loadDocument(KualiDocumentFormBase kualiDocumentFormBase) {
+        super.loadDocument(kualiDocumentFormBase);
+        
+        PaymentRequestDocument document = (PaymentRequestDocument) kualiDocumentFormBase.getDocument();
+        WorkflowDocument workflowDocument = document.getDocumentHeader().getWorkflowDocument();
+        
+        if (workflowDocument.isInitiated() || workflowDocument.isSaved() || workflowDocument.isEnroute()

```

Comment:
could this be simplified to is not finalized and perhaps another place for extraction to avoid duplicates

Reply:
See my comment on the other related spot for a potential change I could make to simplify this.

I'll also look into consolidating this logic if feasible.